howtos/tech/build-cmocka.rst

```diff
@@ -0,0 +1,135 @@
+.. _build-cmocka:
+
+Build Cmocka for Xtensa
+#######################
+
+Cmocka for SOF is built automatically by default.
+However sometimes you may need to have prebuilt version that can be used with :ref:`CMOCKA_DIRECTORY <cmocka-directory-label>`.
+
+This article exaplains how to build Cmocka manually.
+
+Cmocka fork
+***********
+
+We use our Cmocka fork that adds some options for embedded compilers:
+
+WITH_POSITION_INDEPENDENT_CODE
+   Some compilers cannot compile CMake's test program with PIC,
+   so you can disable it with this option.
+
+WITH_TINY_CONFIG
+   Usually compiler checks are cheap, so configuration is quick.
+   However many compilers are expensive to call,
+   so this option can be used to minimize checks count.
+
+WITH_SHARED_LIB
+   For compilers that cannot build shared libs and still want to
+   use make / make install for making prebuilt libs
+
+All changes made to Cmocka are enabled with these options, without them it works just like vanilla Cmocka.
+
+Clone repo and enter its directory, because all examples will be executed there:
+
+   .. code-block:: bash
+
+      git clone https://github.com/thesofproject/cmocka
+      cd cmocka
+
+Simple build on Linux
+*********************
+
+On any system that CMake identifies as Unix, you can just call following
```

Comment:
Line 41-42: "... identifies as Unix-like, you can use the following commands:"

Reply:
Thing is that I literally meant that it sets this variable to true: https://cmake.org/cmake/help/latest/variable/UNIX.html

I'll make it more specific